### ISSUE TITLE
Uncaught (in promise) DOMException: Could not start video source (on Android and using react-media-devices)

### ISSUE THREAD

**Issue by matthiasprieth** [nomadoda/react-zxing#4]
Hi, 
sorry for bothering, but I have another problem when using `react-zxing` and `react-media-devices` together.
I wanted to implement a "Camera switch" and I'm using a similar code as in the documentation.
"react-media-devices": "1.1.0"
"react-zxing": "1.0.4",
```
import { useMediaDevices } from "react-media-devices";
import { useZxing } from "react-zxing";

const constraints: MediaStreamConstraints = {
  video: true,
  audio: false,
};

export const BarcodeScanner = () => {
  const { devices, loading } = useMediaDevices(constraints);
  //... 
  //here I filter the devices 
  //...
  const { ref } = useZxing({
    deviceId: devices[videoDeviceIndexFromLocalStorage].deviceId,
  });

  return <video ref={ref} />;
};
```
So whenever I use `useMediaDevices`(to get the list of available devices) before `react-zxing` it leads to the following error ONLY on Android devices:
![image](https://user-images.githubusercontent.com/4509289/206238195-16ca255c-4549-47fc-8e93-b0bfda926fd5.png)

It seems that Android devices are typically limited to one open camera at a time like described here and that `useMediaDevices` blocks a stream, video or track that `react-zxing` needs:
https://stackoverflow.com/questions/53139293/domexception-could-not-start-video-source-with-zxing-and-bootstrap-4-modal

I've tried a few things to initialize `useMediaDevices` differently but I can't, cause it's a hook & I've tried a few suggestions regarding closing the media streams after getting the devices, but It didn't work for me: https://stackoverflow.com/questions/11642926/stop-close-webcam-stream-which-is-opened-by-navigator-mediadevices-getusermedia

Any ideas on how to solve that? Thx!

**Comment by adamkrupke** [comment]
Hm tricky! I'm not really sure what the reason for this is. I'm reading [this](https://github.com/zxing-js/ngx-scanner/issues/487#issuecomment-1235971152) and [this comment](https://stackoverflow.com/a/53139948/7776268) that seems to suggest there is an issue instantiating a media stream on a video element if there already is one. It would be interesting to know - does the `deviceId`  update?

**Comment by matthiasprieth** [comment]
Hey @nomadoda ,
the `deviceId` stays the same, but for example when I try to access the `deviceId` after initializing a default `useZxing` instance, `devices`/`deviceId` is null (cause the stream/video is in use already i guess). 
```
const { ref } = useZxing({
    onResult(result) {
      setResult(result.getText());
    },
});
const { devices, loading } = useMediaDevices(constraints);
// here devices is null cause it can't access the mediastream anymore or something like that
```
So there must be a way to release the media stream/ video element after getting the devices with `useMediaDevices`. (Cause I need the devices information only once, I don't need a persistent connection with the stream). I've tried something like that, but I don't know if I'm accessing the right stream here or how it is supposed to work. So it did not work and I'm getting the "could not start video source" error again:

```
const cancelTracks = (stream: MediaStream) => {
    stream.getTracks().forEach((track) => track.stop());
};
const cancelStream = async (constraints?: MediaStreamConstraints) => {
    const stream = await navigator.mediaDevices.getUserMedia(constraints);
    cancelTracks(stream);
};
const { devices, loading } = useMediaDevices(constraints);
// cache devices/deviceId's somewhere in the meantime
cancelStream(constraints);
const { ref } = useZxing({
  deviceId: devices[videoDeviceIndexFromLocalStorage].deviceId,
});
```

**Comment by adamkrupke** [comment]
My guess is that since `deviceId` is null when `useZxing` is called, it sets up a stream using the default constraints. Then, when the `deviceId` is defined, it attempts to start a new stream with that device and doesn't close the previous one properly (or in time).

Maybe it could be solved by having a way to start/stop the decoding, so that you can wait for your `deviceId` before starting the stream. I'll add this, it makes sense to have this option either way.

A workaround could perhaps be to put the `useZxing` + `video` element in a separate component and wait for the `deviceId` in an outer component before rendering it.

**Comment by adamkrupke** [comment]
I've updated to address this in 1.0.7. See if it helps to update and follow [this example](https://github.com/nomadoda/react-zxing#with-specific-device-id).

**Comment by matthiasprieth** [comment]
Hey, 
sorry for my late reply.
Thx so much for your effort!

I fixed it by using 1.0.7 with the `paused` property + a component separation of `react-media-devices` & `react-zxing` with the `<video>` element.
I made sure `<video>` is just being rendered, when the outer `deviceId` is 100% ready.
